datastream-cloud-storage/src/main/java/com/linkedin/datastream/cloud/storage/io/ScribeParquetAvroConverter.java

```diff
@@ -0,0 +1,268 @@
+package com.linkedin.datastream.cloud.storage.io;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import java.util.ListIterator;
+
+import org.apache.avro.Schema;
+import org.apache.avro.SchemaBuilder;
+import org.apache.avro.generic.GenericArray;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.nifi.serialization.record.RecordSchema;
+
+/**
+ * ScribeParquetAvroConverter handles the conversion of avro to parquet compatible avro schema
+ */
+public class ScribeParquetAvroConverter {
+
+  /**
+   * Converts field avro schema to parquet compatible format and handles special cases like timestamp, Arrays, Records.
+   * @param isNotNullable  Whether the field is required or not
+   * @param fieldSchema    Schema of the field
+   * @param field          Field
+   * @param fieldAssembler FieldAssembler
+   * @return
+   */
+  public static SchemaBuilder.FieldAssembler<Schema> getFieldAssembler(Boolean isNotNullable, Schema fieldSchema, Schema.Field field, SchemaBuilder.FieldAssembler<Schema> fieldAssembler) {
+    String typeName = fieldSchema.getType().name();
+    if (typeName.equalsIgnoreCase("ARRAY")) {
+      // For list of nested objects
+      if (fieldSchema.getElementType().getType().getName().equalsIgnoreCase("RECORD")) {
+        Schema elementTypeSchema = generateParquetStructuredAvroSchema(fieldSchema.getElementType(), fieldSchema.getElementType().getName());
+        fieldAssembler.name(field.name())
+            .doc(field.doc())
+            .type()
+            .optional()
+            .array().items(elementTypeSchema);
+      } else {
+        getFieldAssemblerForRecord(isNotNullable, fieldSchema, field, fieldAssembler);
+      }
+    } else if (typeName.equalsIgnoreCase("RECORD")) {
+      // For testing using ProductManufacturerPart, replace with header name in prod
+      if (fieldSchema.getName().equalsIgnoreCase("ScribeHeader") || fieldSchema.getName().equalsIgnoreCase("ProductManufacturerPart")) {
```

Comment:
Need to decide if we are going with splitting the header object. Remove `ProductManufacturerPart` before moving to prod.

Reply:
For now are are going with exploded the scribeHeader object